### PR TITLE
feat: add spec for checkout sessions/secure fields [TA-2252]

### PR DESCRIPTION
**Pull request by thomas-gr4vy**
# Description

This PR adds specifications for the following endpoints:
- `POST` /checkout/sessions
- `GET` /checkout/sessions/{id}
- `DELETE` /checkout/sessions/{id}
- `PUT` /checkout/sessions/{id}/fields

This PR updates the following specifications to add `checkout-session` as an acceptable payment method:
- `POST` /transactions
- `POST` /payment-methods

Re [TA-2252](https://gr4vy.atlassian.net/browse/TA-2252)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/openapi.yaml [hunk 1]
```diff
@@ -215,6 +220,24 @@ paths:
     get:
       $ref: "./paths/card_scheme_definitions__list-card-scheme-definitions.yaml"
 
+  "/checkout/sessions":
+    post:
+      $ref: "./paths/checkout_session__add-session.yaml"
```
Comment (cbetta) [inline reference/openapi.yaml:225]: @thomas-gr4vy these paths are generally the plural. E.g. `/checkout-sessions`

File: reference/paths/checkout_session__add-session.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+operationId: add-checkout-session
+
+summary: Create a new Checkout Session
+description: Creates a new Checkout Session.
+tags:
+  - Checkout Sessions
+
+responses:
+  "201":
+    description: Returns details about a current Checkout Session.
```
Comment (cbetta) [inline reference/paths/checkout_session__add-session.yaml:12]:
```suggestion
    description: Returns the new Checkout Session.
```

File: reference/paths/checkout_session_fields__update-fields.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+---
+
+operationId: update-checkout-session-fields
+
+summary: Update a Checkout Session's Secure Fields
+description: Updates the Secure Fields of the Checkout Session.
```
Comment (cbetta) [inline reference/paths/checkout_session_fields__update-fields.yaml:6]:
```suggestion
description: |-
  Updates the Secure Fields of the Checkout Session. Once the fields have been
  received the `expiration_date` will be updated to 5 minutes from the time of 
  receipt.
```

**Review by cbetta — CHANGES_REQUESTED** [review]
@thomas-gr4vy few small improvements and questions.

File: reference/paths/checkout_session_fields__update-fields.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  - Checkout Sessions
+
+requestBody:
+  content:
+    application/json:
+      schema:
+        $ref: "../openapi.yaml#/components/schemas/CheckoutSessionSecureFieldsUpdate"  # yamllint disable-line rule:line-length
+
+responses:
+  "200":
```
[the remark this answers is not in the document]
Comment (thomas-gr4vy) [reply reference/paths/checkout_session_fields__update-fields.yaml:17]: We return a `200` for all other update endpoints. For example:
https://github.com/gr4vy/gr4vy-openapi/blob/8b9c8164e223a59682302733128407b5e28a3cd8/reference/paths/buyers__update-buyer.yaml#L20:L27

File: reference/request-bodies/checkout_sessions_fields_update_request.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        description: |-
+          The 3 or 4 digit security code often found on the card. This often
+          referred to as the CVV or CVD.
+
+          If the `number` of this card represents a tokenized card, then
+          this value is ignored.
+        pattern: '^\d{3,4}$'
+        minLength: 3
+        maxLength: 4
+        example: "123"
```
[the remark this answers is not in the document]
Comment (thomas-gr4vy) [reply reference/request-bodies/checkout_sessions_fields_update_request.yaml]: I have updated this to reference the existing `CardRequest`. This adds properties like `buyer_id` and `buyer_external_identifier`, however, as we are likely to re-use the same schema internally then these will be valid. Do we think this is OK?

File: reference/paths/checkout_sessions__delete-session.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+---
+
+operationId: delete-checkout-session
+
+summary: Delete a Checkout Session
+description: Delete a Checkout Session.
```
Comment (luca-gr4vy) [inline reference/paths/checkout_sessions__delete-session.yaml:6]:
```suggestion
description: Deletes a Checkout Session.
```

I guess, if we're going for third person in the description, for consistency's sake

**Review by cbetta — APPROVED** [review]

**Comment by cbetta** [comment]
@thomas-gr4vy all LGTM

**Review by cbetta — APPROVED** [review]

**Pull request merged**